tests/Integration/TestCase.php

```diff
@@ -176,4 +176,46 @@ public function go_to_new_post(): int {
 
 		return $post_id;
 	}
+
+	/**
+	 * Simulate a single outbound HTTP request and response.
```

Comment:
How does WP core tests handle mocking remote network requests?

Reply:
> How does WP core tests handle mocking remote network requests?

From this search:
https://github.com/WordPress/wordpress-develop/search?q=pre_http_request

...it looks like it's just manually hooking into `pre_http_request` and not unhooking (I guess it relies on some sort of isolation of the test setup).  That could work here as well, I suppose.